### PR TITLE
Support JA (Japanese 🇯🇵 ) language

### PR DESCRIPTION
**Pull request by pdcp1**
This PR adds support for the Japanese language.

More information about adding a new language in our [wiki page](https://github.com/responsible-ai-collaborative/aiid/wiki/Adding-a-new-language-support)

## Important notes before merge it
- `ja` available language is temporary harcoded in `site/gatsby-site/config.js`

## Testing preview link 🇯🇵 
https://pr-2562--staging-aiid.netlify.app/ja/

### CODE REVIEW

**Comment by technohippy** [comment]
Sorry, I'm not sure what I did but changes in my branch may be disappeared with these commands:

```
$ git merge master   # mistakenly I did
$ git push origin feature-support-ja
$ git reset --hard ORIG_HEAD
$ git merge staging
```

Luckily your branch holds everything and more. Could you take over my changes? I don't need my credit. I just want to have Japanese version of AIID.

**Comment by kepae** [comment]
Sure thing @technohippy! We're using your contribution to learn how we can expand and better maintain our language infrastructure in the future – it's uncovered a lot of rough points – and we want people to be able to contribute more easily. Thanks for working with us.

Also, if you would like to keep up a conversation outside of GitHub, we have an active Slack that we use to discuss development details, bugs, and other AIID agenda. Please feel welcome to join (and lurk):
<p >
  <a href="https://forms.gle/v7UHJvEkYSJQ7jHj7"></a>
</p>

**Comment by technohippy** [comment]
@kepae Thanks, I sent an invitation reqest for the slack.

**Comment by pdcp1** [comment]
@kepae @cesarvarela this is ready for review

File: site/gatsby-site/cypress/e2e/integration/citeEdit.cy.js
Comment (cesarvarela) [inline site/gatsby-site/cypress/e2e/integration/citeEdit.cy.js:749]: Japanese looking familiar now

File: site/gatsby-site/src/templates/post.js
Comment (cesarvarela) [inline site/gatsby-site/src/templates/post.js:17]: I like the `const` change, but the other stuff seems unnecessary 🤔

Comment (pdcp1) [reply site/gatsby-site/src/templates/post.js:17]: The reason for this change is to improve the variable's name clarity using the word `translated`. I will change the code to use a comment instead.

**Comment by kepae** [comment]
Overall, LGTM! I started looking into how we can merge this new language and translate future reports only, without translating all of the old historical ones (for now). It's an expense we haven't budgeted for.

In either this PR or a separate PR to merge before this one, what do you think about adding a configuration variable to query and translate reports with a `date_submitted` after a given (validated) date?

e.g.
```js

class Translator {
  constructor({
    mongoClient,
    translateClient,
    languages,
    submissionDateStart = process.env.TRANSLATE_SUBMISSION_DATE_START,  // New argument or config variable, whatever is cleaner
    reporter,
    // TODO: rollback this temporary hack
    //dryRun = process.env.TRANSLATE_DRY_RUN !== 'false',
    dryRun = true,
  }) {
...
// Validate submissionDateStart...
...
{ date_submitted: { $gte: submissionDateStart } }
```

https://github.com/responsible-ai-collaborative/aiid/blob/478ba2c326c8ef78672ec02bc0b5ff24469cba48/site/gatsby-site/src/utils/Translator.js#L130

For me, this is the last and only blocker (translation cost of 3000+ reports 🤯 ).

**Review by cesarvarela — APPROVED** [review]

**Comment by kepae** [comment]
Let's not merge this until we've configured the `TRANSLATE_SUBMISSION_DATE_START` behavior, please.

**Comment by pdcp1** [comment]
@kateperk I removed the hardcoded value on `GATSBY_AVAILABLE_LANGUAGES` variable.
Before merging this PR we should set the `TRANSLATE_SUBMISSION_DATE_START` variable with a specific date with format `yyyy-mm-dd` and the `GATSBY_AVAILABLE_LANGUAGES` with value `en,es,fr,ja`

**Comment by kepae** [comment]
I've configured `TRANSLATE_SUBMISSION_DATE_START` in the staging and production environments.
To update the actions and pull in this variable, I created https://github.com/responsible-ai-collaborative/aiid/pull/2699 -- which might need to hit the master branch before it takes effect in staging...

**Comment by pdcp1** [comment]
> I've configured `TRANSLATE_SUBMISSION_DATE_START` in the staging and production environments. To update the actions and pull in this variable, I created #2699 -- which might need to hit the master branch before it takes effect in staging...

Sounds good!

**Comment by kepae** [comment]
Set in staging:
* `TRANSLATE_SUBMISSION_DATE_START=2024-03-01`
* `GATSBY_AVAILABLE_LANGUAGES=en,es,fr,ja`

And #2699 is in production.

@pdcp1 ready to merge to staging? We can confirm that we're not expected to translate reports older than march now?

**Review by kepae — APPROVED** [review]

**Comment by pdcp1** [comment]
Let's go! 🚀

**Pull request merged**
